replica/table.cc

```diff
@@ -2859,31 +2860,31 @@ db::replay_position table::highest_flushed_replay_position() const {
 
 future<>
 table::seal_snapshot(sstring jsondir, std::vector<snapshot_file_set> file_sets) {
-    std::ostringstream ss;
-    int n = 0;
-    ss << "{" << std::endl << "\t\"files\" : [ ";
-    for (const auto& fsp : file_sets) {
-      for (const auto& rf : *fsp) {
-        if (n++ > 0) {
-            ss << ", ";
-        }
-        ss << "\"" << rf << "\"";
-        co_await coroutine::maybe_yield();
-      }
-    }
-    ss << " ]" << std::endl << "}" << std::endl;
+    struct manifest_json : public json::json_base {
+        json::json_list<sstring> sstable_data_files;
 
-    auto json = ss.str();
+        manifest_json() {
+            add(&sstable_data_files, "files", true);
+        }
+    };
+    
     auto jsonfile = jsondir + "/manifest.json";
 
     tlogger.debug("Storing manifest {}", jsonfile);
 
     co_await io_check([jsondir] { return recursive_touch_directory(jsondir); });
     auto f = co_await open_checked_file_dma(general_disk_error_handler, jsonfile, open_flags::wo | open_flags::create | open_flags::truncate);
     auto out = co_await make_file_output_stream(std::move(f));
+    manifest_json manifest;
     std::exception_ptr ex;
     try {
-        co_await out.write(json.c_str(), json.size());
+        for (const auto& fsp : file_sets) {
+            for (const auto& rf : *fsp) {
+                manifest.sstable_data_files.push(rf);
+                co_await coroutine::maybe_yield();
+            }
+        }
+        co_await manifest.write(out);
```

Comment:
The original plan was to stream the json contents into the output file, but the above method seems to be doing a first round of `push`es that store all file names in the manifest.sstable_data_files vector, and then writes them out in `manifest.write(out)`.